src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -802,6 +804,50 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+// 210824 Touch of the Magi (Aura) - SPELL_MAGE_TOUCH_OF_THE_MAGI_AURA
+class spell_mage_touch_of_the_magi_aura : public AuraScript
+{
+    PrepareAuraScript(spell_mage_touch_of_the_magi_aura);
+
+    uint32 _sum = 0;
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MAGE_TOUCH_OF_THE_MAGI_EXPLODE });
+    }
+
+    void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        DamageInfo* damageInfo = eventInfo.GetDamageInfo();
+        if (damageInfo)
+            if (damageInfo->GetAttacker() == GetCaster() && damageInfo->GetVictim() == GetTarget())
+            {
+                uint32 extra = CalculatePct(damageInfo->GetDamage(), 25);
+                if (extra > 0)
+                {
+                    _sum += extra;
+                    AuraEffect* auraEffect = GetTarget()->GetAuraEffect(SPELL_MAGE_TOUCH_OF_THE_MAGI_AURA, 0, GetCaster()->GetGUID());
+                    auraEffect->ChangeAmount(_sum);
```

Comment:
`auraEffect == aurEff`

Reply:
Is that smart? what if down the line we decide aurEff argument will be one that actually shouldn't be changed?
Otherwise - what would you say about removing the const from the function signature?

Reply:
But your code is doing the same - since both point to the same AuraEffect object

Reply:
the result is the same currently because the const that is passed is the same as the one from `GetAuraEffect` and like you said, it doesn't have to be const. But in the future if someone will want to change the variable that is passed, they won't go looking into each script and check no one is doing a const_cast.

Either way `const_cast` isn't something we should do. Like I said, if it's the same I can remove the const from the signature, assuming you agree.

Reply:
Remove const from argument, I agree.